pkg/token/token-util/token-util_test.go

```diff
@@ -0,0 +1,127 @@
+	"github.com/stretchr/testify/require"
+)
+
+func TestDecodeAccessToken(t *testing.T) {
+	type args struct {
+		rawToken string
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    Token
+		wantErr bool
+	}{
+		{
+			name: "Decode access token",
+			args: args{
+				rawToken: "eyJhbGciOiJSUzI1NiIsInR5cCIgOiAiSldUIiwia2lkIiA6ICJGd1BrQlJUbWY2Rm54R045SFp2VVhmOGhGa0xiOVI4" +
+					"am9XSWVORzNSNlJZIn0.eyJqdGkiOiJkZjRlYjg1OS0zOTA2LTRmNDktOGRlZi0wMzA5MjJiZTRkMDYiLCJleHAiOjE1NTc5Nj" +
+					"I3NjQsIm5iZiI6MCwiaWF0IjoxNTU3OTQ4MzY0LCJpc3MiOiJodHRwczovL2lkcC1icm9rZXIuZGV2LmhwZWRldm9wcy5uZXQv" +
+					"YXV0aC9yZWFsbXMvbWFzdGVyIiwiYXVkIjpbIm1hc3Rlci1yZWFsbSIsImFjY291bnQiXSwic3ViIjoiMTBhYzIxZDQtNTg5Yi" +
+					"00M2NhLWE0MzQtYWQxOTk5ZjM5NTBhIiwidHlwIjoiQmVhcmVyIiwiYXpwIjoib25lc3BoZXJlLWF1dGgiLCJub25jZSI6IjEy" +
+					"MzQ1IiwiYXV0aF90aW1lIjoxNTU3OTQ4MzYzLCJzZXNzaW9uX3N0YXRlIjoiOGI1YTY1ZGYtODkwOS00ZmE3LWJmNWEtYWQ2ZT" +
+					"M5M2Y0ZDVlIiwiYWNyIjoiMSIsImFsbG93ZWQtb3JpZ2lucyI6WyJodHRwczovL2lkcC1icm9rZXIuZGV2LmhwZWRldm9wcy5u" +
+					"ZXQiXSwicmVhbG1fYWNjZXNzIjp7InJvbGVzIjpbImNyZWF0ZS1yZWFsbSIsIm9mZmxpbmVfYWNjZXNzIiwiYWRtaW4iLCJ1bW" +
+					"FfYXV0aG9yaXphdGlvbiJdfSwicmVzb3VyY2VfYWNjZXNzIjp7Im1hc3Rlci1yZWFsbSI6eyJyb2xlcyI6WyJ2aWV3LWlkZW50" +
+					"aXR5LXByb3ZpZGVycyIsInZpZXctcmVhbG0iLCJtYW5hZ2UtaWRlbnRpdHktcHJvdmlkZXJzIiwiaW1wZXJzb25hdGlvbiIsIm" +
+					"NyZWF0ZS1jbGllbnQiLCJtYW5hZ2UtdXNlcnMiLCJxdWVyeS1yZWFsbXMiLCJ2aWV3LWF1dGhvcml6YXRpb24iLCJxdWVyeS1j" +
+					"bGllbnRzIiwicXVlcnktdXNlcnMiLCJtYW5hZ2UtZXZlbnRzIiwibWFuYWdlLXJlYWxtIiwidmlldy1ldmVudHMiLCJ2aWV3LX" +
+					"VzZXJzIiwidmlldy1jbGllbnRzIiwibWFuYWdlLWF1dGhvcml6YXRpb24iLCJtYW5hZ2UtY2xpZW50cyIsInF1ZXJ5LWdyb3Vw" +
+					"cyJdfSwiYWNjb3VudCI6eyJyb2xlcyI6WyJtYW5hZ2UtYWNjb3VudCIsIm1hbmFnZS1hY2NvdW50LWxpbmtzIiwidmlldy1wcm" +
+					"9maWxlIl19fSwic2NvcGUiOiJvcGVuaWQgZW1haWwgdGVuYW50SWQgcHJvZmlsZSIsImVtYWlsX3ZlcmlmaWVkIjp0cnVlLCJu" +
+					"YW1lIjoiQWJoaXNoZWsgU3JpdmFzdGF2YSIsInRlbmFudElkIjoiOHI0MC1sZDlwIiwicHJlZmVycmVkX3VzZXJuYW1lIjoic3" +
+					"JpdmFzdGF2YS5hYmhpc2hla0BocGUuY29tIiwiZ2l2ZW5fbmFtZSI6IkFiaGlzaGVrIiwiZmFtaWx5X25hbWUiOiJTcml2YXN0" +
+					"YXZhIiwiZW1haWwiOiJzcml2YXN0YXZhLmFiaGlzaGVrQGhwZS5jb20ifQ.FP40XhVuO_-1ZsVH4FKzK6HKUig7F1Ahwl2c9RQ" +
+					"CDGp2WxA3xkCOoA_5lFwsVJatCT1f9vRw0LuTDRCIo-8bLv374X8F8V1rThK5ReUBom5-0ul4rruWSL13VTfhgkRhtrjNIwp00" +
+					"IgitVO_mO_hpjusOqQk3uWglpkI1zbFrP5kXaOmy6qj_RLnxS3NCOSdvieEs_r_YN5TuzE75T6OP3_AxFUKpbnOLIH_5TTQtQk" +
+					"oZJfGaIge195FCT5i1o6MqCP3xB0_zfyIbP86lhgyTfyow1SaiDf2uEvtKSrZtmxgJjhAGLWPClSklGX4sky6mnfqNd-ReF5LZ" +
+					"rqEClUAeQ",
+			},
+			want: Token{
+				Issuer:          "https://idp-broker.dev.hpedevops.net/auth/realms/master",
+				Subject:         "users/10ac21d4-589b-43ca-a434-ad1999f3950a",
+				TenantID:        "8r40-ld9p",
+				Expiry:          1557962764,
+				Type:            "Bearer",
+				IssuedAt:        1557948364,
+				Nonce:           "12345",
+				AuthorizedParty: "onesphere-auth",
```

Comment:
The onesphere reference is a bit strange here. Maybe change if the string doesn't matter.

Reply:
Can ignore, not critical.